nixos/modules/system/boot/unl0kr.nix

```diff
@@ -84,60 +81,62 @@ in
 
     boot.initrd.systemd = {
       contents."/etc/unl0kr.conf".source = settingsFormat.generate "unl0kr.conf" cfg.settings;
-      storePaths = with pkgs; [
+      storePaths = (with pkgs; [
         "${pkgs.gnugrep}/bin/grep"
         libinput
         xkeyboard_config
         "${config.boot.initrd.systemd.package}/lib/systemd/systemd-reply-password"
         (lib.getExe' cfg.package "unl0kr")
-      ];
+        
+      ]) ++ ([
+        "${cfg.package}/libexec/unl0kr-agent"
+      ]);
       services = {
-        unl0kr-ask-password = {
-          description = "Forward Password Requests to unl0kr";
+        unl0kr-agent = {
+          description = "Dispatch Password Requests to unl0kr";
```

Comment:
Did you try just adding `buffybox` to `boot.initrd.systemd.packages` rather than duplicating the upstream service things? Would be a lot cleaner.

Reply:
I'm attempting to use the password agent from upstream, rewritten in C.

Reply:
I currently don't know how to debug the systemd service, is there a manual entry for debugging stage 1 services?

Reply:
There should be a way to get a shell in stage 1, then you can manually run commands to check whatever you want to do. Would that be helpful?

Reply:
> I'm attempting to use the password agent from upstream, rewritten in C.

Yes. The buffybox package has upstream unit files that can be included in initrd automatically by setting it in `boot.initrd.systemd.packages` so you don't have to copy it here.

> I currently don't know how to debug the systemd service, is there a manual entry for debugging stage 1 services?

You can get a debug shell in systemd initrd by adding the `rd.systemd.debug_shell` kernel param

Reply:
> > I'm attempting to use the password agent from upstream, rewritten in C.
> 
> Yes. The buffybox package has upstream unit files that can be included in initrd automatically by setting `boot.initrd.systemd.packages = [ buffybox ];` so you don't have to reimplement the units yourself here.
> 
> > I currently don't know how to debug the systemd service, is there a manual entry for debugging stage 1 services?
> 
> You can get a debug shell in systemd initrd by adding the `rd.systemd.debug_shell` kernel param and switching to VT9

After testing and debugging I've noticed that:
- Manually starting the agent on the debug shell makes it work
- The systemd service is inactive (dead)

Reply:
I think you forgot to include `wantedBy` definitions. But again, I think you should delete your custom service and path units and replace them with:

```
boot.initrd.systemd = {
  /* ... */

  packages = [ buffybox ];
  paths.unl0kr-agent.wantedBy = [ "initrd.target" ];
};
```

Reply:
You're right, it works now (although is now victim of a race condition)

Reply:
Or maybe i'll try to see if there's a way for it to force redraws?

Reply:
Anyways, I've made an option to explicitly enable unl0kr's own password agent. I'm unsure as to whether or not just doing away with the old agent (maybe someone from *BSD systems would appreciate that?)

Reply:
(I've also made DRM rendering default to avoid boot logs from shitting out our beautiful disk unlocker)